proc/threads.c

```diff
@@ -1011,7 +1013,7 @@ void proc_threadEnd(void)
 	hal_spinlockSet(&threads_common.spinlock, &sc);
 
 	cpu = hal_cpuGetID();
-	t = threads_common.current[cpu];
+	t = _proc_current();
```

Comment:
Previous version is better (we must access `threads_common.current` directly anyway, one `hal_cpuGetID()` call less)